### PR TITLE
Allow for config reading and fix error in init

### PR DESCRIPTION
**Pull request by LamaAni**

### CODE REVIEW

File: yapenv/cli.py [hunk 1]
```diff
@@ -30,9 +30,7 @@ def env_file(self) -> str:
         return self.get("env_file", os.environ.get("YAPENV_ENV_FILE", ".env"))
 
     def load(
-        self,
-        resolve_imports: bool = True,
-        ignore_environment: bool = False,
+        self, resolve_imports: bool = True, ignore_environment: bool = False, inherit_depth: int = None
```
Comment (phuber92) [inline yapenv/cli.py]: Kinda weird formatting, should be this right?
```suggestion
        self,
        resolve_imports: bool = True,
        ignore_environment: bool = False,
        inherit_depth: int = None
```

Comment (LamaAni) [reply yapenv/cli.py]: Fixed.

File: yapenv/config.py [hunk 2]
```diff
@@ -94,6 +94,14 @@ def pip_install_args(self) -> List[str]:
     def virtualenv_args(self) -> List[str]:
         return self.get("virtualenv_args", [])
 
+    def to_dictionary(self) -> dict:
+        """Convert this config to a dictionary"""
+        return clean_data_types(self)
+
+    def search(self, *paths:str):
+        "Search the config for specific paths"
```
Comment (phuber92) [inline yapenv/config.py]:
```suggestion
        "Search the config for specific paths. i.e. a.b[0].c"
```

File: yapenv/config.py [hunk 3]
```diff
@@ -94,6 +94,14 @@ def pip_install_args(self) -> List[str]:
     def virtualenv_args(self) -> List[str]:
         return self.get("virtualenv_args", [])
 
+    def to_dictionary(self) -> dict:
+        """Convert this config to a dictionary"""
+        return clean_data_types(self)
+
+    def search(self, *paths:str):
+        "Search the config for specific paths. i.e. a.b[0].c"
```
Comment (phuber92) [inline yapenv/config.py]:
```suggestion
        """Search the config for specific dictionary paths. 
        Paths is a list string representations of dictionary paths.
        Ex: 'a.b[0].c'
        """
```

File: yapenv/config.py [hunk 4]
```diff
@@ -94,6 +94,17 @@ def pip_install_args(self) -> List[str]:
...
 
+    def to_dictionary(self) -> dict:
+        """Convert this config to a dictionary"""
+        return clean_data_types(self)
+
+    def search(self, *paths:str):
+        """Search the config for specific dictionary paths. 
+        Paths is a list string representations of dictionary paths.
+        Ex: 'a.b[0].c'
+        """
```
Comment (phuber92) [inline yapenv/config.py]:
```suggestion
        """Search the config for specific dictionary paths. 
        Paths is a list of string representations of dictionary paths.
        Ex: paths = ['a.b[0].c']
        """
```

**Review by phuber92 — DISMISSED** [review]

**Review by phuber92 — APPROVED** [review]

**Pull request merged**